hudi-flink/src/main/java/org/apache/hudi/table/format/cow/Int64TimestampColumnReader.java

```diff
@@ -75,22 +84,38 @@ protected void readBatchFromDictionaryIds(
     for (int i = rowId; i < rowId + num; ++i) {
       if (!column.isNullAt(i)) {
         column.setTimestamp(i, decodeInt64ToTimestamp(
-            utcTimestamp, dictionary, dictionaryIds.getInt(i)));
+            utcTimestamp, dictionary, dictionaryIds.getInt(i), unit));
       }
     }
   }
 
   public static TimestampData decodeInt64ToTimestamp(
       boolean utcTimestamp,
       org.apache.parquet.column.Dictionary dictionary,
-      int id) {
+      int id,
+      LogicalTypeAnnotation.TimeUnit unit) {
     long value = dictionary.decodeToLong(id);
-    return int64ToTimestamp(utcTimestamp, value);
+    return int64ToTimestamp(utcTimestamp, value, unit);
   }
 
-  private static TimestampData int64ToTimestamp(boolean utcTimestamp, long millionsOfDay) {
+  private static TimestampData int64ToTimestamp(
+      boolean utcTimestamp,
+      long millionsOfDay,
+      LogicalTypeAnnotation.TimeUnit unit) {
     if (utcTimestamp) {
-      return TimestampData.fromEpochMillis(millionsOfDay, 0);
+      final ChronoUnit chronoUnit;
+      switch (unit) {
+        case MILLIS:
+          chronoUnit = ChronoUnit.MILLIS;
+          break;
```

Comment:
I guess the `chronoUnit` can be a class member variable right ?
And, we should also fix the non-utc timezone code path.